### PR TITLE
Reorganize and update documentation 📋

### PR DESCRIPTION
**Pull request by hackebrot**
This pull requests is a continuation of the work in #330.

For the purpose of discoverability, I decided to use `/application/README.md` as the home page for application specific documentation and link out to separate files where it makes sense.

### CODE REVIEW

File: application/docs/glean.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    `technology.space_travel` metric
+    4. submit a `discovery` ping regardless of wether an error occured or not
+6. handle any unexpected errors and set an according exit code
+7. wait 5 seconds for telemetry to be sent before terminating the app
+
+## Burnham missions
+
+Burnham missions are defined in [src/burnham/missions.py][missions.py]
+
+As mentioned about for most missions `complete()` performs a sequence of
```
Comment (badboy) [inline application/docs/glean.md:26]:
```suggestion
As mentioned above for most missions `complete_mission()` performs a sequence of
```

File: application/docs/glean.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+`space_ship.jump()` and `space_ship.warp()` calls, which in turn records to the
+`technology.space_travel` metric.
+
+* `MissionE` is special in that it submits a `starbase46` and then produce
```
Comment (badboy) [inline application/docs/glean.md:30]:
```suggestion
* `MissionE` is special in that it submits a `starbase46` ping and then produces
```

File: application/docs/glean.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+## Glean pings
+
+Glean pings are defined in [src/burnham/config/pings.yaml][pings.yaml].
+
+[metrics.yaml]: ../src/burnham/config/metrics.yaml
+[pings.yaml]: ../src/burnham/config/pings.yaml
+[cli.py]: ../src/burnham/cli.py
+[missions.py]: ../src/burnham/missions.py
+
```
Comment (badboy) [inline application/docs/glean.md:50]:
```suggestion
```

**Review by badboy — APPROVED** [review]

File: application/docs/glean.md (hunk 1 above)
Comment (hackebrot) [reply application/docs/glean.md:26]: In this particular case `complete()` is correct as that's the method name on mission classes. Do you mind updating your suggestion to only have the `above` change, @badboy?

**Review by jrbenny35 — APPROVED** [review]
Looks good!

**Review by jklukas — APPROVED** [review]

**Pull request merged**